### PR TITLE
Deploy refactor helm ready 

### PR DESCRIPTION
**Pull request by rcompos**
Helm chart for vault ready.

### CODE REVIEW

File: bin/start.sh [hunk 1]
```diff
@@ -1,5 +1,5 @@
 #!/bin/bash
 . './bin/strict-mode.sh'
-. ./bin/set-env-secrets.sh
+#. ./bin/set-env-secrets.sh
```
Comment (edhedges) [inline bin/start.sh:3]: Is this no longer needed? I thought we saw that it was.

File: helm/bloom-vault/templates/server-deployment.yaml [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+          ports:
+            - containerPort: {{ .Values.server.service.port }}
+              protocol: TCP
+          #livenessProbe:
+          #  httpGet:
+          #    path: /
+          #    port: http
+          #readinessProbe:
+          #  httpGet:
+          #    path: /api/v1/health?app={{ include "bloom-vault.server.fullname" . }}
```
Comment (edhedges) [inline helm/bloom-vault/templates/server-deployment.yaml:46]: IMO we should implement this and uncomment the `readinessProbe` declaration in this PR for completeness. I can take a look at it.

File: helm/bloom-vault/values.yaml [hunk 3]
```diff
@@ -0,0 +1,104 @@
...
+    # lines, adjust them as necessary, and remove the curly braces after 'resources:'.
+    # limits:
+    #   cpu: 100m
+    #   memory: 128Mi
+    # requests:
+    #   cpu: 100m
+    #   memory: 128Mi
+
+  autoscaling:
+    enabled: false
```
Comment (edhedges) [inline helm/bloom-vault/values.yaml:90]: Should this be enabled and if so should the maxReplicas be set to much less?

File: helm/bloom-vault/templates/server-deployment.yaml (hunk 2 above)
Comment (edhedges) [reply helm/bloom-vault/templates/server-deployment.yaml:46]: @rcompos This is done in [this commit](https://github.com/hellobloom/bloom-vault/pull/28/commits/52d56e82b39d66620083492063f7af48c92230f1) and can be incorporated into the helm chart.

Comment (rcompos) [reply helm/bloom-vault/templates/server-deployment.yaml:46]: readiness probe uncommented

File: helm/bloom-vault/values.yaml (hunk 3 above)
Comment (rcompos) [reply helm/bloom-vault/values.yaml:90]: enabled with max of 5

**Comment by rcompos** [comment]
Helm chart:
https://github.com/hellobloom/bloom-vault/helm/bloom-vault

Values.yaml for global cluster in stag namespace:
https://github.com/hellobloom/k8s-infra/global-bloom-vault/values-global-bloom-vault-stag.yaml

**Pull request merged**